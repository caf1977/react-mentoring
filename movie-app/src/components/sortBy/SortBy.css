.sort-by-container {
    display: flex;
    gap: 10px;
    color: #555555;
    font-size: small;
    align-items: center;
}

.sort-by-list {
    background-color: #232323;
    font-size: small;
    color: white;
    border: none;
}

.sort-by-list option {
    font-size: small;
}