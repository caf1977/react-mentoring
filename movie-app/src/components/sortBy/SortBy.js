import React from 'react';
import './SortBy.css';

const SortBy = ({
    currentSelection = "",
    onSelectionChange,
}) => {
    const handleChange = (event) => {
        const newValue = event.target.value;
        onSelectionChange(newValue);
    }

    return (
        <div className="sort-by-container">
            <label htmlFor="sort-by-list">
                SORT BY:
            </label>
            <select
                className="sort-by-list"
                id="sort-by-list"
                value={currentSelection}
                onChange={handleChange}
            >
                <option value="release_date">RELEASE DATE</option>
                <option value="title">TITLE</option>
            </select>    
        </div>
    );
};

export default SortBy;