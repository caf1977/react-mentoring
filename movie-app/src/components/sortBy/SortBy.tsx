import React, { ChangeEvent, FC } from 'react';
import './SortBy.css';

interface ISortBy {
    currentSelection?: string;
    onSelectionChange?: (selection: string) => void;
}

const SortBy: FC<ISortBy> = ({
    currentSelection = "",
    onSelectionChange,
}) => {
    const handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
        const newValue = event.target.value;
        onSelectionChange?.(newValue);
    }

    return (
        <div className="sort-by-container">
            <label htmlFor="sort-by-list">
                SORT BY:
            </label>
            <select
                className="sort-by-list"
                id="sort-by-list"
                value={currentSelection}
                onChange={handleChange}
            >
                <option value="release_date">RELEASE DATE</option>
                <option value="title">TITLE</option>
            </select>    
        </div>
    );
};

export default SortBy;