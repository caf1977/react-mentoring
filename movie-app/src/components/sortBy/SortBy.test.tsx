import React from 'react';
import {fireEvent, render, screen} from "@testing-library/react";
import SortBy from './SortBy';

describe("SortBy component tests", () => {
    it("renders sortBy dropdown with options", () => {
        render(<SortBy />);
        const dropdown = screen.getByLabelText("SORT BY:");

        expect(dropdown).toBeInTheDocument();
        const options = screen.getAllByRole("option");
        expect(options).toHaveLength(2);
        expect(options[0]).toHaveTextContent(/release date/i);
        expect(options[1]).toHaveTextContent(/title/i);
    })

    it("release date as current selection", () => {
        render(<SortBy currentSelection={"release_date"}/>);
        const dropdown = screen.getByLabelText("SORT BY:");

        expect(dropdown).toBeInTheDocument();
        const options = screen.getAllByRole("option") as HTMLOptionElement[];
        expect(options).toHaveLength(2);
        expect(options[0].selected).toBe(true);
        expect(options[1].selected).toBe(false);
    })

    it("title as current selection", () => {
        render(<SortBy currentSelection={"title"}/>);
        const dropdown = screen.getByLabelText("SORT BY:");

        expect(dropdown).toBeInTheDocument();
        const options = screen.getAllByRole("option") as HTMLOptionElement[];
        expect(options).toHaveLength(2);
        expect(options[0].selected).toBe(false);
        expect(options[1].selected).toBe(true);
    })

    it("onChange is called when option is selected", () => {
        window.alert = jest.fn();
        render(<SortBy onSelectionChange={(value) => {alert(value)}}/>);
        const dropdown = screen.getByLabelText("SORT BY:");
        fireEvent.click(dropdown);
        fireEvent.change(dropdown, { target: {value: "title"}});

        expect(window.alert).toHaveBeenCalledWith("title");
    })

    it("renders snapshot", () => {
        const { asFragment } = render(<SortBy currentSelection={"release_date"}/>);

        expect(asFragment()).toMatchSnapshot();
    })
})