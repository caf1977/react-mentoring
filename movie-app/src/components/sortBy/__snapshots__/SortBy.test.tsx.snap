// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy component tests renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="sort-by-container"
  >
    <label
      for="sort-by-list"
    >
      SORT BY:
    </label>
    <select
      class="sort-by-list"
      id="sort-by-list"
    >
      <option
        value="release_date"
      >
        RELEASE DATE
      </option>
      <option
        value="title"
      >
        TITLE
      </option>
    </select>
  </div>
</DocumentFragment>
`;
